sdk/state/payment.go

```diff
@@ -352,3 +352,34 @@ func (c *Channel) ConfirmPayment(ce CloseEnvelope) (closeAgreement CloseAgreemen
 
 	return c.latestAuthorizedCloseAgreement, nil
 }
+
+// FinalizePayment finalizes a payment, making it authorized, by attaching the
+// close signatures to the agreement as the confirmers signatures. The proposer
+// of a payment calls this once with the confirmers signatures when the
+// confirmer provides them. This can only be used to finalize the most recent
+// unauthorized payment.
+func (c *Channel) FinalizePayment(cs CloseSignatures) (closeAgreement CloseAgreement, err error) {
```

Comment:
💭 this can be only be called by the proposer right? if so, can we rename to `FinalizeProposedPayment`? imo I think makes it clearer who calls + when the method should be called

Reply:
Good point. It's not clear today that both participants call Confirm. The good thing about the API is that a mistake calling the wrong function has no unsafe affect. Any incorrect call will cause an error rather than break the channel, but we should do better to lead the developer for sure.

I think introducing the word Proposed could still be misleading because a receiver could think of the close agreement they've received as a proposed agreement, and could think they need to finalize the proposed agreement. So I don't think the name change addresses the overarching problem with understanding what functions to call when. 

I've pushed some new docs in 68c671dee1f6c916532f8f30945a658b002346ff. I think this can be the start of more comprehensive docs for the code. What do you think?

Reply:
> I think introducing the word Proposed could still be misleading because a receiver could think of the close agreement they've received as a proposed agreement

good point. The case I was thinking of is the receiver receives a new agreement and thinks they can "finalize" it by calling `FinalizePayment` instead of `ConfirmAgreement`. The diagram you added makes it pretty clear though that they should call Confirm, so lgtm 👍  if they do call Finalize by mistake it fails anyways